src/dbnode/encoding/m3tsz/timestamp_iterator.go

```diff
@@ -195,7 +193,7 @@ func (it *TimestampIterator) tryReadMarker(stream encoding.IStream) (time.Durati
 	}
 }
 
-func (it *TimestampIterator) readMarkerOrDeltaOfDelta(stream encoding.IStream) (time.Duration, error) {
+func (it *TimestampIterator) readMarkerOrDeltaOfDelta(stream encoding.IStream) (xtime.UnixNano, error) {
```

Comment:
Similarly here can you just keep the return of this method `readDeltaOfDelta` as `time.Duration`?